src/components/page-deliveries/page-deliveries.tsx

```diff
@@ -350,28 +372,18 @@ export class PageDeliveries {
                 <ul>
                   {selectedAddress != null ? (
                     <OrderAndTruckInfoList
-                      items={
-                        locationItems.length > 0
-                          ? locationItems.slice(0, 3)
-                          : // show placeholders if no data
-                            [
-                              { type: "pizza", data: null },
-                              { type: "pizza", data: null },
-                              { type: "pizza", data: null },
-                            ]
-                      }
+                      items={locationItems.length > 0 ? locationItems.slice(0, 3) : []}
                       selectedLocation={selectedLocation}
-                      onOrderSelected={order => (this.selectedOrder = this.recentOrders?.find(x => x.id === order.id))}
+                      onOrderSelected={order => (this.selectedOrder = order)}
                     />
                   ) : (
-                    (items?.length > 0 ? items?.slice(0, 3) : ([null, null, null] as (OrderOrTruckItem | null)[])) // show placeholders if no data
-                      .map(x =>
-                        x != null && x.type === "truck" ? (
-                          <TruckInfoDisplay truck={x.data} onClick={() => this.selectLocation(x?.data?.location)} />
-                        ) : (
-                          <OrderDetailDisplay order={x?.data} onClick={() => this.selectLocation(x?.data?.location)} />
-                        ),
-                      )
+                    (currentItems?.length > 0 ? currentItems : []).map(x =>
+                      x != null && x.type === "truck" ? (
```

Comment:
This could be `x?.type === "truck"` since `(undefined)?.type !== "truck"`

Reply:
For whatever reason the static type resolver wasn't able to detect the type of `x` in the else condition correcty when I used `x?.type === "truck"`